### DISCUSSION TITLE
Using Immich CLI to create albums based on External Library

### DISCUSSION THREAD

**Discussion by srahimeen** [immich-app/immich#13899] [Q&A] [answered]
I use Immich as a way to view my Google Takeout backups, and have added a `Google Photos` dir as an External Library on Immich. It has indexed all the files works great, however I would like to re-create the albums as well. Luckily Google Takeout provides the images in album-named folders as well.

I was looking into using the Immich CLI to create the albums based on the folder names in my `Google Photos` dir. I tried the following to see how it would work:

```
PS Z:\Google Photos\Plants> immich upload --dry-run --album .
Crawling for assets...
Checking files | ████████████████████████████████████████ | 100% | ETA: 0s | 7/7 assets
Found 7 new files and 0 duplicates
Would have uploaded 7 assets (9.5 MB)
Would have created 1 new album
Would have updated albums of 7 assets
```

However, it seems like running this would re-upload the images in this album folder to Immich local library. I was expecting the CLI to find duplicates as these images are already on Immich via the External library, but it seems like the hashing function does not consider the External Library natively. I don't want to re-upload all my images to the Immich local library as it would lead to two copies of the same files on my server.

Is there some way to use the Immich CLI to:
1. Create albums based on the folder name
2. Associate the images in the folder to the created album, but without re-uploading them

Basically a way to work with images which are already on Immich via an External Library, but just need to be "added" to albums (and not freshly uploaded).

Does anyone know of a way to make this process work? Thanks!

**Comment by bo0tzz** [comment, marked as answer]
The CLI is for uploads and external libraries are not considered in the upload duplicate detection. There's some community projects that do what you want, you'll find some on https://immich.app/docs/community-projects.

**Reply by srahimeen** [reply]
Thanks for the info! This script did the trick for me: https://github.com/Salvoxia/immich-folder-album-creator

The command to run was:
```
python ./immich_auto_album.py "/usr/src/app/external" http://192.168.1.130:2283/api API_KEY --ignore Untitled
```
